src/coreclr/tools/aot/ILCompiler.ReadyToRun/CodeGen/ReadyToRunObjectWriter.cs

```diff
@@ -456,10 +456,13 @@ private void EmitObjectData(R2RPEBuilder r2rPeBuilder, ObjectData data, int node
                 {
                     Console.WriteLine($@"Duplicate symbol - 1st occurrence: [{alreadyWrittenSymbol.NodeIndex}:{alreadyWrittenSymbol.SymbolIndex}], {alreadyWrittenSymbol.Node.GetMangledName(_nodeFactory.NameMangler)}");
                     Console.WriteLine($@"Duplicate symbol - 2nd occurrence: [{nodeIndex}:{symbolIndex}], {definedSymbol.GetMangledName(_nodeFactory.NameMangler)}");
-                    Debug.Fail("Duplicate node name emitted to file",
-                    $"Symbol {definedSymbol.GetMangledName(_nodeFactory.NameMangler)} has already been written to the output object file {_objectFilePath} with symbol {alreadyWrittenSymbol}");
+                 //   Debug.Fail("Duplicate node name emitted to file",
```

Comment:
This looks like a useful assertion. Did you intend to leave it commented out?

Reply:
Sigh, yes, I'll turn this assert back on. I found a place where our name mangler wasn't completely competent at generating unique names. As I recall, it caused problems compiling Roslyn with the optimizations entirely enabled, but it was two method fixups which were very subtly distinct in their behavior.

Reply:
It's not a problem for crossgen in reality, as duplicate symbols are not actually a fault, but as we share a name mangler with NativeAOT it's probably worth chasing down sooner rather than later.